mobile-academy/src/main/java/org/motechproject/nms/mobileacademy/service/impl/MobileAcademyServiceImpl.java

```diff
@@ -0,0 +1,133 @@
+package org.motechproject.nms.mobileacademy.service.impl;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.joda.time.DateTime;
+import org.motechproject.mtraining.domain.Bookmark;
+import org.motechproject.mtraining.repository.BookmarkDataService;
+import org.motechproject.nms.mobileacademy.domain.Course;
+
+import org.motechproject.nms.mobileacademy.dto.MaBookmark;
+import org.motechproject.nms.mobileacademy.repository.CourseDataService;
+import org.motechproject.nms.mobileacademy.service.MobileAcademyService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Simple implementation of the {@link MobileAcademyService} interface.
+ */
+@Service("mobileAcademyService")
+public class MobileAcademyServiceImpl implements MobileAcademyService {
+
+    /**
+     * Bookmark data service
+     */
+    private BookmarkDataService bookmarkDataService;
+
+    /**
+     * Course data service
+     */
+    private CourseDataService courseDataService;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MobileAcademyServiceImpl.class);
+
+    @Autowired
+    public MobileAcademyServiceImpl(BookmarkDataService bookmarkDataService,
+                                    CourseDataService courseDataService) {
+        this.bookmarkDataService = bookmarkDataService;
+        this.courseDataService = courseDataService;
+    }
+
+    @Override
+    public Course getCourse() {
+
+        // Make this course name configurable
+        String lookupName = "MobileAcademyCourse";
+        return courseDataService.findCourseByName(lookupName);
+    }
+
+    @Override
+    public void setCourse(Course course) {
+        Course existing = courseDataService.findCourseByName(course.getName());
+
+        if (existing == null) {
+            courseDataService.create(course);
+        } else {
+            course.setId(existing.getId());
+            courseDataService.update(course);
+        }
+    }
+
+    @Override
+    public int getCourseVersion() {
+
+        Course course = getCourse();
+        if (course != null) {
+            DateTime version = course.getModificationDate();
+            return (int) version.getMillis();
+        } else {
+            // return -1 and let the caller handle the upstream response
+            return -1;
+        }
+    }
+
+    @Override
+    public MaBookmark getBookmark(Long callingNumber, Long callId) {
+
+        List<Bookmark> bookmarks = bookmarkDataService.findBookmarksForUser(callingNumber.toString());
+        if (CollectionUtils.isEmpty(bookmarks)) {
+            return null;
+        } else {
+
+            if (bookmarks.size() > 1) {
+                LOGGER.debug("Found more than 1 instance of valid course, picking top");
```

Comment:
Should this read "Found than one instance of valid bookmark..." rather than course?